### DISCUSSION TITLE
Java version does not change

### DISCUSSION THREAD

**Discussion by microwin7** [community/community#58148] [Codespaces]
### Select Topic Area

Bug

### Body

I installed java, as I always did on linux
```sudo apt-get update ; sudo apt-get install gnupg2 wget apt-transport-https -y ; wget -q -O - https://download.bell-sw.com/pki/GPG-KEY-bellsoft | sudo apt-key add - ; echo "deb [arch=amd64] https://apt.bell-sw.com/ stable main" | sudo tee /etc/apt/sources.list.d/bellsoft.list ; sudo apt-get update ; sudo apt-get install -y bellsoft-java17-full```

But I can't change. And it is not clear where 11 came from when it is not on the list

```
@microwin7 ➜ /workspaces/Launcher (master) $ java -version
openjdk version "11.0.14.1" 2022-02-08 LTS
OpenJDK Runtime Environment Microsoft-31205 (build 11.0.14.1+1-LTS)
OpenJDK 64-Bit Server VM Microsoft-31205 (build 11.0.14.1+1-LTS, mixed mode)

@microwin7 ➜ /workspaces/Launcher (master) $ sudo update-alternatives --config java
There is only one alternative in link group java (providing /usr/bin/java): /usr/lib/jvm/bellsoft-java17-full-amd64/bin/java
Nothing to configure.

@microwin7 ➜ /workspaces/Launcher (master) $ java -version
openjdk version "11.0.14.1" 2022-02-08 LTS
OpenJDK Runtime Environment Microsoft-31205 (build 11.0.14.1+1-LTS)
OpenJDK 64-Bit Server VM Microsoft-31205 (build 11.0.14.1+1-LTS, mixed mode)
```

Since I need to build via gradle, I had to do this:
```./gradlew build -Dorg.gradle.java.home=/usr/lib/jvm/bellsoft-java17-full-amd64```

**Comment by joshspicer** [comment]
Hey 👋 Is there a `devcontainer.json` file in your repo?  The conflicting versions of java might be coming from that config. If there is no `devcontainer.json` you should get the default image [which installs java 17 and 11 from the Microsoft feed](https://github.com/devcontainers/images/blob/main/src/universal/.devcontainer/devcontainer.json#L44-L49).  

On a repo I have without any `devcontainer.json`, I get java version 17 by default:

```
@joshspicer ➜ /workspaces/rustlings (main) $ java --version
openjdk 17.0.7 2023-04-18 LTS
OpenJDK Runtime Environment Microsoft-7626293 (build 17.0.7+7-LTS)
OpenJDK 64-Bit Server VM Microsoft-7626293 (build 17.0.7+7-LTS, mixed mode, sharing)
```

If you don't want any of the default JDKs bundled and want to install it yourself via that `apt` command you shared, I'd suggest starting with a more minimal dev container/image, perhaps using one of the two options below, creating following at least a `.devcontainer/devcontainer.json`.

## Option 1:  Using on Create Command and your `apt` commands

#### .devcontainer/devcontainer.json
```json
{
   "image": "mcr.microsoft.com/devcontainers/base:ubuntu",
   "onCreateCommand": ".devcontainer/onCreate.sh"
}
```
#### .devcontainer/onCreate.sh
```shell
#!/bin/bash

sudo apt-get update ; sudo apt-get install gnupg2 wget apt-transport-https -y ; wget -q -O - https://download.bell-sw.com/pki/GPG-KEY-bellsoft | sudo apt-key add - ; echo "deb [arch=amd64] https://apt.bell-sw.com/ stable main" | sudo tee /etc/apt/sources.list.d/bellsoft.list ; sudo apt-get update ; sudo apt-get install -y bellsoft-java17-full
```


## Option 2:  Using Features

```
{
   "image": "mcr.microsoft.com/devcontainers/base:ubuntu",
   "features": {
      "ghcr.io/devcontainers/features/java:1": {
          "version": "17"
       }
    }
}
```

Using Features will execute the install scripts from here, and also install tools like SDKMAN. That may or may not be what you're looking for!

https://github.com/devcontainers/features/tree/main/src/java

**Reply by microwin7** [reply]
Okay, I'll try what you said, I created the Codespace from the code on GitHub automatically and could only choose which machine to use for this. The configuration file that you wrote is not yet familiar to me, I will study this issue. I just have a problem just more with java in that I don’t just need version 17, but I need it with JavaFX, that’s why I usually use this command for a quick installation

**Reply by microwin7** [reply]
I suspect, since I used the creation of codespace from the code on GitHub, and there is mainly java code, this container for java was used. This is where version 11 is listed.
https://github.com/devcontainers/images/blob/5366f3b9b99bf81f612bfc4c59c3030aa79a2e14/src/java/.devcontainer/Dockerfile#LL2C27-L2C27
